src/crypto/CHIPCryptoPAL.cpp

```diff
@@ -188,6 +256,33 @@ CHIP_ERROR Spake2p::ComputeRoundOne(uint8_t * out, size_t * out_len)
     VerifyOrExit(state == CHIP_SPAKE2P_STATE::STARTED, error = CHIP_ERROR_INTERNAL);
     VerifyOrExit(*out_len >= point_size, error = CHIP_ERROR_INTERNAL);
 
+#if ((CHIP_CRYPTO_HSM) && (!ENABLE_HSM_SPAKE_VERIFIER))
```

Comment:
I find these ifdefs hard to follow. Could we ifdef entire function implementations instead ? Alternatively have some helper `IsMissingImplementation(role)` and narrow ifdefs in there.

However really it seems we have eitherHSM implementation or nonHSM.

Reply:
Do you have a suggestion here? If we ifdef entire functions, wouldn't it  be too much code duplication. For the second approach, as the ifdefs wraps different codes at different places, IsMissingImplementation(role) would mean 'different' at different places. Is my understanding correct or am I missing it?

Reply:
I found it hard to judge how much is duplicated - at a first glance it seemed the code paths were completely separate except some precondition/argument checks. I was hoping that we could have completely separated out logic.

It could be done as a followup if you believe it is impractical. many separate `#ifdef` does make code review difficult.

Please apply `./scripts/helpers/restyle-diff.sh` to resolve restyle errors. This will add more reviewers and the PR will not show up as red in open PRs.

Reply:
Thanks I will apply ./scripts/helpers/restyle-diff.sh. I was not aware of it. Regarding ifdef, We can separate it for sure as a follow-up PR, But one other thing that complicates the separation is that there is fallback option to software at certain places. But will definitely look into it